backend/Shared/HitMeApp.Shared.Infrastructure/Integration/Memory/InMemoryIntegrationEventBus.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using HitMeApp.Shared.Infrastructure.Integration.Subscriptions;
+
+namespace HitMeApp.Shared.Infrastructure.Integration.Memory
+{
+    internal class InMemoryIntegrationEventBus
+    {
+        private static readonly Lazy<InMemoryIntegrationEventBus> _instance = new Lazy<InMemoryIntegrationEventBus>(() => new InMemoryIntegrationEventBus());
+
+        private readonly List<IIntegrationEventSubscription> _subscriptions = new List<IIntegrationEventSubscription>();
+
+        public static InMemoryIntegrationEventBus Instance => _instance.Value;
+
+        private InMemoryIntegrationEventBus()
+        {
+            // Hidden constructor due to singleton design pattern
+        }
+
+        public Task Publish<TIntegrationEvent>(params TIntegrationEvent[] events) where TIntegrationEvent : IntegrationEvent
+            => Publish(events?.AsEnumerable());
+
+        public Task Publish<TIntegrationEvent>(IEnumerable<TIntegrationEvent> events) where TIntegrationEvent : IntegrationEvent
+            => Task.WhenAll(events.Select(@event => Publish(@event)));
+
+        private Task Publish<TIntegrationEvent>(TIntegrationEvent @event) where TIntegrationEvent : IntegrationEvent
+        {
+            var handlers = _subscriptions.Where(subscription => subscription as IntegrationEventSubscription<TIntegrationEvent> is { })
+                .Cast<IntegrationEventSubscription<TIntegrationEvent>>()
+                .Select(subscription => subscription.HandlerResolver().Handle(@event))
+                .ToList();
```

Comment:
`Task.WhenAll` accepts `IEnumerable`. We don't have to cast it to `List`.

Reply:
The signature was different before. Right now it makes no sense to keep it that way. Corrected.